properties/test_index_manipulation.py

```diff
@@ -0,0 +1,156 @@
+import random
+
+import numpy as np
+import pytest
+
+from xarray import Dataset
+from xarray.indexes import PandasMultiIndex
+from xarray.testing import _assert_internal_invariants
+
+pytest.importorskip("hypothesis")
+
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+from hypothesis import assume, note, settings
+from hypothesis.stateful import RuleBasedStateMachine, invariant, precondition, rule
+
+import xarray.testing.strategies as xrst
+
+
+def get_not_multiindex_dims(ds: Dataset) -> set:
+    dims = ds.dims
+    mindexes = [
+        name
+        for name, index in ds.xindexes.items()
+        if isinstance(index, PandasMultiIndex)
+    ]
+    return set(dims) - set(mindexes)
+
+
+def get_multiindex_dims(ds: Dataset) -> list:
+    mindexes = [
+        name
+        for name, index in ds.xindexes.items()
+        if isinstance(index, PandasMultiIndex)
+    ]
+    return mindexes
+
+
+def get_dimension_coordinates(ds: Dataset) -> set:
+    return set(ds.dims) & set(ds._variables)
+
+
+@st.composite
+def unique(draw, strategy):
+    # https://stackoverflow.com/questions/73737073/create-hypothesis-strategy-that-returns-unique-values
+    seen = draw(st.shared(st.builds(set), key="key-for-unique-elems"))
+    return draw(
+        strategy.filter(lambda x: x not in seen).map(lambda x: seen.add(x) or x)
+    )
+
```

Comment:
Presumably you could just use the `xrst.unique_subset_of` strategy with `minsize=1` and `maxsize=1` instead here?

Reply:
ah `unique_subset_of` is indeed useful, but not for this case I don't think. Repeated draws can result in duplicated values, and I wanted to avoid that.